cmd/ttn-lw-cli/commands/use.go

```
@@ -0,0 +1,129 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package commands
+
+import (
+	"crypto/tls"
+	"encoding/pem"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/spf13/cobra"
+	"go.thethings.network/lorawan-stack/pkg/errors"
+	"gopkg.in/yaml.v2"
+)
+
+func destPath(base string, user bool) (string, error) {
+	if !user {
+		return base, nil
+	}
+
+	dir, err := os.UserConfigDir()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(dir, base), nil
+}
+
+var (
+	errNoHost      = errors.DefineInvalidArgument("no_host", "no host set")
+	errFileExists  = errors.DefineAlreadyExists("file_exists", "`file` exists")
+	errFailWrite   = errors.DefinePermissionDenied("fail_write", "failed to write `file`")
+	configFileName = ".ttn-lw-cli.yml"
+	caFileName     = "ca.pem"
+
+	useCommand = &cobra.Command{
+		Use:               "use [host]",
+		Short:             "Use",
+		PersistentPreRunE: preRun(),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			insecure, _ := cmd.Flags().GetBool("insecure")
+			ca, _ := cmd.Flags().GetBool("ca")
+			user, _ := cmd.Flags().GetBool("user")
+			overwrite, _ := cmd.Flags().GetBool("overwrite")
+			credentialsID, _ := cmd.Flags().GetString("credentials-id")
+
+			if len(args) != 1 {
+				return errNoHost
+			}
+
+			host := args[0]
+			conf := MakeDefaultConfig(host, insecure)
+			if credentialsID != "" {
+				conf.CredentialsID = credentialsID
+			}
+
+			// Get CA certificate from server
+			if !insecure && ca {
+				conn, err := tls.Dial("tcp", conf.NetworkServerGRPCAddress, &tls.Config{InsecureSkipVerify: true})
+				if err != nil {
+					return err
+				}
+				defer conn.Close()
+				caFile, err := destPath("ca.pem", user)
+				if err != nil {
+					return err
+				}
+
+				f, err := os.Create(caFile)
```

Do not discard errors